Makefile

```diff
@@ -276,81 +259,96 @@ proto-lint: $(BUF)
 	$(BUF) check breaking --against-input '.git#branch=master'
 
-TM_URL           = https://raw.githubusercontent.com/tendermint/tendermint/v0.34.0-rc3/proto/tendermint
-GOGO_PROTO_URL   = https://raw.githubusercontent.com/regen-network/protobuf/cosmos
-COSMOS_PROTO_URL = https://raw.githubusercontent.com/regen-network/cosmos-proto/master
-COSMOS_SDK_PROTO_URL = https://raw.githubusercontent.com/cosmos/cosmos-sdk/master/proto/cosmos/base
-
-TM_CRYPTO_TYPES     = third_party/proto/tendermint/crypto
-TM_ABCI_TYPES       = third_party/proto/tendermint/abci
-TM_TYPES     	    = third_party/proto/tendermint/types
-TM_VERSION 			= third_party/proto/tendermint/version
-TM_LIBS				= third_party/proto/tendermint/libs/bits
-
-GOGO_PROTO_TYPES    = third_party/proto/gogoproto
-COSMOS_PROTO_TYPES  = third_party/proto/cosmos_proto
-
-SDK_ABCI_TYPES  	= third_party/proto/cosmos/base/abci/v1beta1
-SDK_QUERY_TYPES  	= third_party/proto/cosmos/base/query/v1beta1
-SDK_COIN_TYPES  	= third_party/proto/cosmos/base/v1beta1
-
-proto-update-deps:
-	mkdir -p $(GOGO_PROTO_TYPES)
-	curl -sSL $(GOGO_PROTO_URL)/gogoproto/gogo.proto > $(GOGO_PROTO_TYPES)/gogo.proto
-
-	mkdir -p $(COSMOS_PROTO_TYPES)
-	curl -sSL $(COSMOS_PROTO_URL)/cosmos.proto > $(COSMOS_PROTO_TYPES)/cosmos.proto
-
-	mkdir -p $(TM_ABCI_TYPES)
-	curl -sSL $(TM_URL)/abci/types.proto > $(TM_ABCI_TYPES)/types.proto
-
-	mkdir -p $(TM_VERSION)
-	curl -sSL $(TM_URL)/version/types.proto > $(TM_VERSION)/types.proto
-
-	mkdir -p $(TM_TYPES)
-	curl -sSL $(TM_URL)/types/types.proto > $(TM_TYPES)/types.proto
-	curl -sSL $(TM_URL)/types/evidence.proto > $(TM_TYPES)/evidence.proto
-	curl -sSL $(TM_URL)/types/params.proto > $(TM_TYPES)/params.proto
-
-	mkdir -p $(TM_CRYPTO_TYPES)
-	curl -sSL $(TM_URL)/crypto/proof.proto > $(TM_CRYPTO_TYPES)/proof.proto
-	curl -sSL $(TM_URL)/crypto/keys.proto > $(TM_CRYPTO_TYPES)/keys.proto
-
-	mkdir -p $(TM_LIBS)
-	curl -sSL $(TM_URL)/libs/bits/types.proto > $(TM_LIBS)/types.proto
-
-	mkdir -p $(SDK_ABCI_TYPES)
-	curl -sSL $(COSMOS_SDK_PROTO_URL)/abci/v1beta1/abci.proto > $(SDK_ABCI_TYPES)/abci.proto
+.PHONY: protovendor
+protovendor: modsensure $(MODVENDOR)
+	@echo "vendoring *.proto files..."
+	@$(MODVENDOR) -copy="**/*.proto" -include='\
+github.com/cosmos/cosmos-sdk/proto,\
+github.com/tendermint/tendermint/proto,\
+github.com/gogo/protobuf,\
+github.com/regen-network/cosmos-proto/cosmos.proto'
+
+# Tools installation
+$(CACHE):
+	@echo "creating .cache dir structure..."
+	@mkdir -p $@
+	@mkdir -p $(CACHE_BIN)
+	@mkdir -p $(CACHE_INCLUDE)
+	@mkdir -p $(CACHE_VERSIONS)
 
-	mkdir -p $(SDK_QUERY_TYPES)
-	curl -sSL $(COSMOS_SDK_PROTO_URL)/query/v1beta1/pagination.proto > $(SDK_QUERY_TYPES)/pagination.proto
+$(GOLANGCI_LINT_VERSION_FILE): $(CACHE)
+	@echo "installing golangci-lint..."
+	@rm -f $(CACHE_BIN)/golangci-lint
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
+		sh -s -- -b $(CACHE_BIN) $(GOLANGCI_LINT_VERSION)
+	@rm -rf $(dir $@)
```

Comment:
1. quote arguments everywhere - `"$(dir $@)"`
1. I'd prefer to not have the `@` prefix on anything.

Reply:
`$(NOECHO)rm -f $(CACHE_BIN)/golangci-lint` etc? so by default nothing is suppressed

Reply:
I think that won't work because `@` is `make` syntax, not shell.  I could be wrong.  

It's not a huge issue but it's all inconsistent right now and I prefer to keep it the more verbose version when it's messing around with files.

Reply:
it does 😉 
```sh
❯ make setup-devenv
echo "creating .cache dir structure..."
creating .cache dir structure...
mkdir -p /Users/amr/go/src/github.com/ovrclk/akash/.cache
mkdir -p /Users/amr/go/src/github.com/ovrclk/akash/.cache/bin
mkdir -p /Users/amr/go/src/github.com/ovrclk/akash/.cache/include
mkdir -p /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions
installing golangci-lint...
rm -f /Users/amr/go/src/github.com/ovrclk/akash/.cache/bin/golangci-lint
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
        sh -s -- -b /Users/amr/go/src/github.com/ovrclk/akash/.cache/bin v1.27.0
golangci/golangci-lint info checking GitHub for tag 'v1.27.0'
golangci/golangci-lint info found version: 1.27.0 for v1.27.0/darwin/amd64
golangci/golangci-lint info installed /Users/amr/go/src/github.com/ovrclk/akash/.cache/bin/golangci-lint
rm -rf /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/golangci-lint/
mkdir -p /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/golangci-lint/
touch /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/golangci-lint/v1.27.0
installing protoc buf cli...
rm -f /Users/amr/go/src/github.com/ovrclk/akash/.cache/bin/buf
curl -sSL \
        "https://github.com/bufbuild/buf/releases/download/v0.20.5/buf-Darwin-x86_64" \
        -o "/Users/amr/go/src/github.com/ovrclk/akash/.cache/bin/buf"
chmod +x "/Users/amr/go/src/github.com/ovrclk/akash/.cache/bin/buf"
rm -rf /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/buf/
mkdir -p /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/buf/
touch /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/buf/0.20.5
installing protoc compiler...
rm -f /Users/amr/go/src/github.com/ovrclk/akash/.cache/bin/protoc
(cd /tmp; \
    curl -sOL "https://github.com/protocolbuffers/protobuf/releases/download/v3.11.2/protoc-3.11.2-osx-x86_64.zip"; \
    unzip -oq protoc-3.11.2-osx-x86_64.zip -d /Users/amr/go/src/github.com/ovrclk/akash/.cache bin/protoc; \
    unzip -oq protoc-3.11.2-osx-x86_64.zip -d /Users/amr/go/src/github.com/ovrclk/akash/.cache 'include/*'; \
    rm -f protoc-3.11.2-osx-x86_64.zip)
rm -rf /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/protoc/
mkdir -p /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/protoc/
touch /Users/amr/go/src/github.com/ovrclk/akash/.cache/versions/protoc/3.11.2
installing modvendor...
GOBIN=/Users/amr/go/src/github.com/ovrclk/akash/.cache/bin GO111MODULE=off go get github.com/goware/modvendor
go mod vendor
```

```sh
NOECHO=@ make setup-devenv
creating .cache dir structure...
installing golangci-lint...
golangci/golangci-lint info checking GitHub for tag 'v1.27.0'
golangci/golangci-lint info found version: 1.27.0 for v1.27.0/darwin/amd64
golangci/golangci-lint info installed /Users/amr/go/src/github.com/ovrclk/akash/.cache/bin/golangci-lint
installing protoc buf cli...
installing protoc compiler...
installing modvendor...
go mod vendor
```

Reply:
huh, cool.  according to `make man`, you can use `make -s` for silent mode.  I'd prefer that over adding that variable in front of every line.

Reply:
yah, cool